File: Easy/Second Largest/second-largest.java
//{ Driver Code Starts
//Initial Template for Java
import java.util.stream.Collectors; 
import java.util.*;
import java.io.*;

public class Main {

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int tc = Integer.parseInt(br.readLine().trim());
        while (tc-- > 0) {
            String[] inputLine;
            int n = Integer.parseInt(br.readLine().trim());
            int[] arr = new int[n];
            inputLine = br.readLine().trim().split(" ");
            for (int i = 0; i < n; i++) {
                arr[i] = Integer.parseInt(inputLine[i]);
            }

            int ans = new Solution().print2largest(arr, n);
            System.out.println(ans);
        }
    }
}

// } Driver Code Ends


//User function Template for Java

class Solution {
    int print2largest(int arr[], int n) {
        // code here
          // code here
        int m1=arr[0];
        int res=1;
       int m2=0;
        for(int i=1;i<n;i++){
            if(arr[i]>m1){
                m2=m1;
                m1=arr[i];
            }
            else if(m2<arr[i] && arr[i]<m1 )
            {
                m2=arr[i];
            }
             if(arr[i-1]==arr[i])
            {
                res++;
            }
            
        }
       if(res==n){return -1;}
        return m2;
    }
}

File: Difficulty: Easy/Second Largest/second-largest.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;


// } Driver Code Ends
// User function template for C++
class Solution {
  public:
    // Function returns the second
    // largest elements
    int print2largest(int arr[], int n) {
        // code here
        int mx=-1;
        int mn=-1;
        for(int i=0;i<n;i++){
            if(mx<arr[i]){
                mn=mx;
                mx=arr[i];
            }
            if(arr[i]>mn && arr[i]<mx){
                mn=arr[i];
            }
        }
        return mn;
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        int arr[n];
        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }
        Solution ob;
        auto ans = ob.print2largest(arr, n);
        cout << ans << "\n";
    }
    return 0;
}

// } Driver Code Ends